tests/filecheck/dialects/onnx/onnx_invalid.mlir

```diff
@@ -484,3 +484,38 @@ builtin.module {
 
 }
 
+// -----
+
+builtin.module {
+  %t0, %t1 = "test.op"() : () -> (tensor<2x4xf32>, tensor<4x2xi8>)
+
+  // CHECK: Operation does not verify: operands must have the same type
+  %res_matmul =  "onnx.MatMul"(%t0, %t1) {onnx_node_name = "/MatMul"} : (tensor<2x4xf32>, tensor<4x2xi8>) -> tensor<2x2xf32>
+}
+
+// -----
+
+builtin.module {
+  %t0, %t1 = "test.op"() : () -> (tensor<2x4x3xf32>, tensor<4x2xf32>)
+
+  // CHECK: Operation does not verify: matrix A must be a 2D tensor
+  %res_matmul =  "onnx.MatMul"(%t0, %t1) {onnx_node_name = "/MatMul"} : (tensor<2x4x3xf32>, tensor<4x2xf32>) -> tensor<2x2xf32>
+}
+
+// -----
+
+builtin.module {
+  %t0, %t1 = "test.op"() : () -> (tensor<2x4xf32>, tensor<4x2x3xf32>)
+
+  // CHECK: Operation does not verify: matrix B must be a 2D tensor
+  %res_matmul =  "onnx.MatMul"(%t0, %t1) {onnx_node_name = "/MatMul"} : (tensor<2x4xf32>, tensor<4x2x3xf32>) -> tensor<2x2xf32>
+}
+
+// -----
+
+builtin.module {
+  %t0, %t1 = "test.op"() : () -> (tensor<2x4xf32>, tensor<5x2xf32>)
+
+  // CHECK: Operation does not verify: matrix A shape[1] must be equal to matrix B shape[0]
+  %res_matmul =  "onnx.MatMul"(%t0, %t1) {onnx_node_name = "/MatMul"} : (tensor<2x4xf32>, tensor<5x2xf32>) -> tensor<2x2xf32>
+}
```

Comment:
```suggestion
}

```

Reply:
Done

Reply:
It's back :) You might want to configure your editor to always add a trailing newline to your files